### ISSUE TITLE
[Bug] Doesn't Remove Check-Marked Mission

### ISSUE THREAD

**Issue by EvilsSouls** [SebLague/Geographical-Adventures#33]
After dropping a delivery off with a reasonable score it doesn't remove the check-marked mission.

![bugged_geography_thing](https://user-images.githubusercontent.com/86242809/166300031-a653b307-9272-47ae-a6d1-177d8ac4f344.png)

I'm using Linux Mint.

**Comment by selwynorren** [comment]
Hmm Yup I had this happen once as well. I will try and see if I can replicate

**Comment by EvilsSouls** [comment]
> Hmm Yup I had this happen once as well. I will try and see if I can replicate

Yeah... For me it happens almost every time when I drop a delivery off. I first thought that it only happened when I had a reasonable score, but now it seems like just when I drop something off.

**Comment by SebLague** [comment]
If you're willing to open the project in the Unity editor and run it there, it'd be very helpful to know if it's throwing some sort of error when this issue occurs.

**Comment by a96lex** [comment]
Hi, I just noticed I created a duplicate here #37, I'll close it.
I noticed that this happened when i dropped a package on map view (#36)

**Comment by dis-da-moe** [comment]
> Hi, I just noticed I created a duplicate here #37, I'll close it. I noticed that this happened when I dropped a package on map view (#36)

I've been trying to replicate this issue, so thanks for saying that it happens when map view is open. 

@SebLague Unity produces this error when the UI is meant to update:

![Screenshot 2022-05-03 000133](https://user-images.githubusercontent.com/84386186/166327103-e7ff2c8e-4e9e-4f45-b4d1-25b2fab8b2c7.png)

I can do a PR that stops packages being dropped from the map view, it should solve this issue and issue #36

**Comment by SebLague** [comment]
Thanks for investigating! I've fixed the package/map bug in the latest commit (will update builds soon). Can anyone confirm if this issue can still arise without having first attempted to drop a package in map view?

**Comment by Sobgo** [comment]
I'm still able to recreate this issue by opening map while swap animation is being played.
Old name doesn't disappear and new name gets stuck on screen edge (shorter names can be even completely off-screen)
![unknown](https://user-images.githubusercontent.com/60814050/166507304-499df17a-0edb-40cd-ae71-0413d242e334.png)
then after next animation names will overlap
![unknown2](https://user-images.githubusercontent.com/60814050/166508033-5c0c9ba1-3df0-4266-a928-4cb14b808c0e.png)

**Comment by Sobgo** [comment]
I don't know Unity specifics but one solution to this could be checking/repositioning all city names after closing a map.

**Comment by SebLague** [comment]
Thanks @Sobgo, I've been able to reproduce the issue now and will work on a fix.

**Comment by EvilsSouls** [comment]
> Thanks @Sobgo, I've been able to reproduce the issue now and will work on a fix.

I downloaded the newest version on itch and I again got the issue.
![image](https://user-images.githubusercontent.com/86242809/166560084-f04ec4e2-dc97-47da-879c-ec349b0021cc.png)

I as normal opened the map after I dropped the package off and it happend. Although this has been the first time again since like 1.5 hours. After the bug my mouse also appeared.
